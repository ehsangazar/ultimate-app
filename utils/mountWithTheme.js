import React from 'react'
import { ThemeProvider } from 'emotion-theming'
import renderer from 'react-test-renderer'
import theme from '../configs/theme'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}

export default mountWithTheme
