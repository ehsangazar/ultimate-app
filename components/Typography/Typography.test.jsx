import React from 'react'
import serializer from 'jest-emotion'
import mountWithTheme from '../../utils/mountWithTheme'
import { H1, H2, H3, H4, H5, Paragraph, Small, Space } from '.'

expect.addSnapshotSerializer(serializer)

describe('Typography', () => {
  test('H1 Renders Correctly', () => {
    const component = mountWithTheme(<H1>Heading</H1>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H2 Renders Correctly', () => {
    const component = mountWithTheme(<H2>Heading</H2>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H3 Renders Correctly', () => {
    const component = mountWithTheme(<H3>Heading</H3>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H4 Renders Correctly', () => {
    const component = mountWithTheme(<H4>Heading</H4>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('H5 Renders Correctly', () => {
    const component = mountWithTheme(<H5>Heading</H5>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Paragraph Renders Correctly', () => {
    const component = mountWithTheme(<Paragraph>Heading</Paragraph>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Small Renders Correctly', () => {
    const component = mountWithTheme(<Small>Heading</Small>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Space Renders Correctly', () => {
    const component = mountWithTheme(<Space />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
