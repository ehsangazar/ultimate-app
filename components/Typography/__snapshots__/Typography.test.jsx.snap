// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography H1 Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  font-weight: 500;
  line-height: 1.6;
  margin-top: 0;
  font-size: 1.802em;
}

<h1
  className="emotion-0"
>
  Heading
</h1>
`;

exports[`Typography H2 Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  font-weight: 500;
  line-height: 1.6;
  margin-top: 0;
  font-size: 1.602em;
}

<h1
  className="emotion-0"
>
  Heading
</h1>
`;

exports[`Typography H3 Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  font-weight: 500;
  line-height: 1.6;
  margin-top: 0;
  font-size: 1.424em;
}

<h3
  className="emotion-0"
>
  Heading
</h3>
`;

exports[`Typography H4 Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  font-weight: 500;
  line-height: 1.6;
  margin-top: 0;
  font-size: 1.266em;
}

<h4
  className="emotion-0"
>
  Heading
</h4>
`;

exports[`Typography H5 Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  font-weight: 500;
  line-height: 1.6;
  margin-top: 0;
  font-size: 1.125em;
}

<h5
  className="emotion-0"
>
  Heading
</h5>
`;

exports[`Typography Paragraph Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  margin-bottom: 0.7rem;
  font-weight: 300;
  line-height: 1.6;
  font-size: 1em;
}

<p
  className="emotion-0"
>
  Heading
</p>
`;

exports[`Typography Small Renders Correctly 1`] = `
.emotion-0 {
  text-align: right;
  font-weight: 100;
  line-height: 1.6;
  font-size: 0.889em;
}

<span
  className="emotion-0"
>
  Heading
</span>
`;

exports[`Typography Space Renders Correctly 1`] = `
.emotion-0 {
  width: 100%;
  height: 1em;
}

<div
  className="emotion-0"
/>
`;
